src/getValues.js

```diff
@@ -24,7 +32,7 @@ const getValue = (field, state, dest) => {
         getValue(rest, item, dest[key][index]);
       });
     } else {
-      dest[key] = array.map(item => item && item.value);
+      dest[key] = array.map(itemToValue);
```

Comment:
I think this may need to be updated to

```diff
-.map(itemToValue)
+.map(item => item && itemToValue(item))
```

All tests are currently passing, but I'm not familiar enough with array fields to know off hand if we are missing an edge case. See [should get values from array form](https://github.com/erikthedeveloper/redux-form/blob/bug/621-values-not-initializing/src/__tests__/getValues.spec.js#L73-L89) for current coverage.